### PR TITLE
[RTC-231] move server notifications to protobuffs

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

File: compile_proto.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# git submodule update --recursive --init >> /dev/null
+# printf "DONE\n\n"
+
+files=$(find protos/jellyfish -name "*.proto")
+
+printf "Compiling:\n"
+count=1
+total=${#files[@]}
+for file in $files; do
+  printf "[%i/%i] %s ... " $count $total $file
```
Comment (mickel8) [inline compile_proto.sh:27]: This always prints [1/1]. Should it?

image

File: .formatter.exs [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-  inputs: [
-    "{lib,test,config}/**/*.{ex,exs}",
-    ".formatter.exs",
-    "*.exs"
-  ]
+  inputs:
+    Enum.flat_map(
+      ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
+      &Path.wildcard(&1, match_dot: true)
+    ) -- Path.wildcard("lib/protos/**/*.*", match_dot: true)
```
Comment (mickel8) [inline .formatter.exs:6]:
```suggestion
  inputs: [
    "{lib,test,config}/**/*.{ex,exs}",
    ".formatter.exs",
    "*.exs"
  ],
  import_deps: [:protobuf]
  ```

**Review by mickel8 — CHANGES_REQUESTED** [review]
A couple of minor comments but overall looks good :+1:  

If we decided to change proto API, I would consider moving from tuples to structs :thinking:

File: compile_proto.sh [hunk 3]
```diff
@@ -13,18 +13,21 @@ for dep in $deps; do
...
 count=1
 total=${#files[@]}
 for file in $files; do
-  printf "[%i/%i] %s ... " $count $total $file
+  printf "Compile file %s %s ... " $count $file
   protoc --elixir_out=./lib/ $file
   printf "DONE\n"
-done
\ No newline at end of file
+  count=$(expr $count + 1)
```
Comment (mickel8) [inline compile_proto.sh:30]: FYI, you can do `count=$((count + 1))`

https://askubuntu.com/questions/385528/how-to-increment-a-variable-in-bash

File: compile_proto.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Terminate on errors
+set -e
+
+deps="printf find git protoc"
```
[the remark this answers is not in the document]
Comment (mickel8) [reply compile_proto.sh]: I meant, I don't see why we should check against those commands. Can they be absent?

File: examples/server_socket.exs [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
 
-{:ok, pid} = Jellyfish.Notifier.start(server_address: "localhost:4000", server_api_token: server_api_token)
+{:ok, pid} =
+  Jellyfish.Notifier.start(server_address: "localhost:4000", server_api_token: server_api_token)
 
-receive do
-  {:jellyfish, server_notification} ->
-    IO.inspect(server_notification, label: :server_notification)
+defmodule Receiver do
+  def receive_notification() do
```
Comment (mickel8) [inline examples/server_socket.exs:11]: I would write this as an anonymous function and save it under `receive_notification`

```elixir
receive_notification = fn ->
 receive do
   ...
  end
  receive_notification.()
end

receive_notification.()
```

File: lib/jellyfish/notifier.ex [hunk 6]
```diff
@@ -123,76 +124,35 @@ defmodule Jellyfish.Notifier do
   end
 
   defp decode_notification(%ControlMessage{
-         content: {type, %PeerNotification{roomId: room_id, peerId: id}}
+         content: {_type, %PeerDisconnected{room_id: room_id, peer_id: id}}
```
Comment (mickel8) [inline lib/jellyfish/notifier.ex:127]: I wonder about returning those structs, instead of tuples. Any thoughts?

Comment (LVala) [reply lib/jellyfish/notifier.ex:127]: Seems like a fine idea.

**Review by mickel8 — APPROVED** [review]

**Pull request merged**